pdf/model/colorspace.go

```diff
@@ -509,14 +509,17 @@ func (this *PdfColorspaceDeviceCMYK) ColorToRGB(color PdfColor) (PdfColor, error
 	y := cmyk.Y()
 	k := cmyk.K()
 
-	c = c*(1-k) + k
-	m = m*(1-k) + k
-	y = y*(1-k) + k
-
+	// Peter Williams. I need a cmyk->rgb conversion that gives r!=g or g!=b when c!=m or m!=y
```

Comment:
The above encoding discussion is about color page detection.

In grayscale conversion, I would like to preserve CCITTDecode, JBIG2Decode and RunLengthDecode images because they are known to be grayscale already.

In neither the color page detection nor the grayscale conversion case do I need to decode or encode